plugins/services/src/js/reducers/serviceForm/Containers.js

```diff
@@ -0,0 +1,153 @@
+import {SET, ADD_ITEM, REMOVE_ITEM} from '../../../../../../src/js/constants/TransactionTypes';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+import {combineReducers, simpleFloatReducer} from '../../../../../../src/js/utils/ReducerUtil';
+
+const containerReducer = combineReducers({
+  cpus: simpleFloatReducer('resources.cpus'),
+  mem: simpleFloatReducer('resources.mem'),
+  disk: simpleFloatReducer('resources.disk')
+});
+
+function containers(state, {type, path = [], value}) {
+  if (!path.includes('containers')) {
+    return state;
+  }
+
+  if (this.cache == null) {
+    this.cache = [];
+  }
+
+  if (!state) {
+    state = [];
+  }
+
+  let newState = state.slice();
+  const index = path[1];
+  const joinedPath = path.join('.');
+
+  if (joinedPath === 'containers') {
+    switch (type) {
+      case ADD_ITEM:
+        newState.push({name: `container ${newState.length + 1}`});
+        this.cache.push({});
+        break;
+      case REMOVE_ITEM:
+        newState = newState.filter((item, index) => {
+          return index !== value;
+        });
+        this.cache = this.cache.filter((item, index) => {
+          return index !== value;
+        });
+        break;
+    }
+
+    return newState;
+  }
+
+  if (path[2] === 'artifacts') {
+    if (newState[path[1]].artifacts == null) {
+      newState[path[1]].artifacts = [];
+    }
+
+    switch (type) {
+      case ADD_ITEM:
+        newState[path[1]].artifacts.push('');
+        break;
+      case REMOVE_ITEM:
+        newState[path[1]].artifacts = newState[path[1]].artifacts.filter((item, index) => {
+          return index !== value;
+        });
+        break;
+      case SET:
+        newState[path[1]].artifacts[path[3]] = value;
```

Comment:
Please adjust this reducer and respective parser as well to output/parse the use the correct artifact scheme.
```JSON
{
        "uri": "http://download.me/file.tgz",
        "extract": true,
        "executable": true,
        "cache": false
}
```
For details please see: https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/artifact.raml